python/py-cryptography/Portfile

```diff
@@ -1,49 +1,167 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
+PortGroup           github 1.0
 PortGroup           python 1.0
-PortGroup           compiler_blacklist_versions 1.0
 
-name                py-cryptography
-version             2.9.2
+github.setup        pyca cryptography 3.4.7
+name                py-${github.project}
 revision            0
 categories-append   devel
 platforms           darwin
 license             BSD
 
-python.versions     27 35 36 37 38 39
+python.versions     27 36 37 38 39
 
 maintainers         {stromnov @stromnov} openmaintainer
 
-description         Package which provides cryptographic recipes and primitives to Python developers.
-long_description    ${description}
+description         cryptography is a package designed to expose \
+                    cryptographic primitives and recipes to Python developers.
 
-homepage            https://github.com/pyca/cryptography
+long_description    {*}${description}
 
-checksums           rmd160  61fb86fd231710d5875428a3a1cefcd5c1d0c99e \
+checksums           ${distname}${extract.suffix} \
+                    rmd160  f579f749a23d076f7204bd6cf66d8dceb2ab3866 \
+                    sha256  9eb1a420f9f50e131f451b9dc8950c4f3b2a1298ee9d6e25ba873d3f66354a80 \
+                    size    35737283
+
+# https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst
+set cryptography_min_ver 14
+
+if {${name} ne ${subport}
+    && [string match "py*-${github.project}" ${subport}]} {
+    depends_build-append \
+                port:py${python.version}-setuptools
+
+    depends_lib-append \
+                path:lib/libssl.dylib:openssl \
+                port:py${python.version}-cffi \
+                port:py${python.version}-six
+
+    # legacy support
+    if {${python.version} eq 27
+        || ${os.platform} eq "darwin" && ${os.major} < ${cryptography_min_ver}} {
+        version     2.9.2
+
+        description Legacy support of Python cryptography.
+
+        long_description    {*}${description}
+
+        master_sites \
+                    pypi:[string index ${python.rootname} 0]/${python.rootname}
+        distname    ${python.rootname}-${version}
```

Comment:
why are you switching now to PyPI for this part and not use GitHub as well?

Reply:
Now using git for everything, thanks.